### DISCUSSION TITLE
Is it okay to install undici in Node and use EnvHttpProxyAgent and setGlobalDispatcher?

### DISCUSSION THREAD

**Discussion by bruce-c-liu** [nodejs/undici#3557] [Q&A] [answered]
Hi, I'm trying to use native Node fetch, but I need to set a global proxy dispatcher that respects HTTPS_PROXY env variable. 

So I tried directly installing `undici`:
```js
import { EnvHttpProxyAgent, setGlobalDispatcher } from "undici";
setGlobalDispatcher(new EnvHttpProxyAgent());

fetch(<some url>); // <== notice I'm using Node's native fetch.
```

This seems to work! My question is are there any issues with this approach?

**Comment by mcollina** [comment, marked as answer]
None, that's the way it's done. If there are issues, we'd consider those as bugs.

**Reply by bruce-c-liu** [reply]
Gotcha. 

I've seen folks setting the global dispatcher using this method as well: https://github.com/nodejs/undici/discussions/2167#discussioncomment-6239992

Can you comment on which way is the Node team's recommendation?
